### PR TITLE
[New] Add videojs-background w/ npm auto-update

### LINKED ISSUE #9815: [Request] Add videojs-Background
**Library name:** videojs-Background
**Git repository url:** https://github.com/matthojo/videojs-Background
**License(s):** MIT

For beginners who is interested in picking up this issue:

In this issue, we are going to host a new web front-end library on CDNJS, so that web developer can easily use it on our free CDN without additional download/upload process 🚀 .

If you'll like to work locally, I'll suggest you start pulling cdnjs repository with sparseCheckout: https://github.com/cdnjs/cdnjs/blob/master/documents/sparseCheckout.md, since there are so many files in this repo, without sparseCheckout, the whole process will be slow and will use more than 80GB disk space, using git command line can learn more skills and experience, if you don't like that, instead, you can also just do it on GitHub, it's more convenient, but it may be a little bit harder to fix problems on GitHub gui.

Here are the docs about how to add a library:
 - https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#d-adding-a-new-library-with-its-assets
 - https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#e-adding-a-new-library-by-a-single-packagejson

We'll also ask to setup auto-update config so that the added libraries can be updated to the latest version easily:
 - https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md

For beginners who is interested in join and help add this library, please feel free to tell me what kind of help you'll need, we'll help you send the valid pull request, thanks 😄 

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/39681440-request-add-videojs-background?utm_campaign=plugin&utm_content=tracker%2F32893&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F32893&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by sashberd**
close #9815, cc @matthojo

Pull request for issue: #9815 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/matthojo/videojs-Background/tree/master
 * Official website (optional, not the repository):
 * NPM package url (optional): https://www.npmjs.com/package/videojs-background
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 8
   - Count of stars:66
   - Count of forks:34
 * NPM download stats (optional):
   - Downloads in the last day: 0
   - Downloads in the last week: 16
   - Downloads in the last month: 99

# Essential checklist
 * [ ] I'm the author of this library
   * [X] I would like to add link to the page of this library on CDNJS on website / readme
 * [X] This lib was not found on cdnjs repo
 * [X] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [ ] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 800 downloads stats per month on npm registry
 * [X] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [X] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [X] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [X] The parent of the commit(s) in the PR is not old than 3 days.
 * [X] Pull request is sending from a non-master branch with meaningful name.
 * [X] Separate unrelated changes into different commits.
 * [X] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [X] Close corresponding issue in commit message
 * [X] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by sashberd** [comment]
@sufuf3 I have some troubles with previous PR of this lib. I had to close it and reopen new PR. FUR

**Comment by sashberd** [comment]
@sufuf3  The PR was updated. FYR

**Comment by sashberd** [comment]
@sufuf3 ping

**Comment by sashberd** [comment]
@sufuf3 @extend1994 Ping

File: ajax/libs/videojs-background/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      "files": [
+        "**/*"
+      ]
+    },
+    {
+      "basePath": "lib",
+      "files": [
+        "**/*"
+      ]
+    }
```
Comment (MattIPv4) [inline ajax/libs/videojs-background/package.json]: By expanding these both to the root of the package, we are going to get a clash as files are named the same things in these two dirs:

```
├── dist
│   ├── videojs-background.js
│   └── videojs-background.min.js
├── lib
│   └── videojs-Background.js
```

File: ajax/libs/videojs-background/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      "basePath": "dist",
+      "files": [
+        "**/*"
+      ]
+    },
+    {
+      "basePath": "lib",
+      "files": [
+        "**/*"
+      ]
```
Comment (MattIPv4) [inline ajax/libs/videojs-background/package.json]:
```suggestion
      "basePath": "",
      "files": [
        "lib/**/*"
      ]
```

I would suggest we expand `dist` to the root of the package, but then leave `lib` in its directory.

**Pull request merged**